### DISCUSSION TITLE
Create new object with Selectize field widget

### DISCUSSION THREAD

**Discussion by rez0n** [jrief/django-formset#83] [Q&A]
Hi @jrief 
Thanks for this project, looks very inspiring, would like move from django-select2.

After reading docs I have a question in regard Selectize, are there any way to process creating of new object if no results found for search string?

Explaining:
User inputs search string, get's `No results found for 'xxxxx'` but anyway confirms it, this text being added as selected option into the field, after form submitting custom function in view creates new object based on user input.

PS. [Select2 Dynamic option creation](https://select2.org/tagging)

**Comment by jrief** [comment]
The underlying code of TomSelect can handle this, so in theory it's be possible. I actually never saw a use case in this and it only can work on a Django `ModelChoiceField` but not on a `ChoiceField`.

We then have a permission problem (who is allowed to add another option), etc.

What's your exact use case for this?

**Reply by rez0n** [reply]
Hi @jrief 
Thanks for your answer.
In my case I can't say it is very widely use case, but actually I have like 5 forms in project which needs this exact behaviour.

First of all and simplest example it is tagging some entity - user need pick or create new tag(s) for some entity.

Another example, we have booking system, customer submit booking request for services in some location, then in back-office "sales team" process submitted bookings and confirms by selecting which specific contractor will handle this "booking".
In case if contractor does not exists in the database - empty "placeholder" will be created and which will gives basic information to customer.
Afterward "accounting team" will fill other details about this contractor, but on first stage it could be just a name.

What about permissions.
In my case it is easy, everybody who have access to this specific form have permission to create new entities.
In more complex case it can be solved in form __init__ method.

```python
self.fields['contractor'].widget = Selectize(is_writable=self.request.user.has_perm('app.perm'), ...)
```
Or even by perlacing widget

```python
if self.request.user.has_perm('app.perm'):
    self.fields['contractor'].widget = SelectizeWritable()
else:
    self.fields['contractor'].widget = Selectize()
```

**Comment by jrief** [comment]
Just replacing the widget would not help. To implement this, I must create a `MutableModelChoiceField` and a `MutableMultipleModelChoiceField` with appropriate widgets. This field then will be able to add extra options to the named model.

Where would you add the interface to remove options?

**Reply by rez0n** [reply]
Hi @jrief 
> Where would you add the interface to remove options?

Do you mean removing options which was created using this field? I have no this functionality because it is not needed, but if it will be required in some day - it will be dedicated management page.
